### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by AntonioHincapie**
- Interface
  - Create a class Nameable.
    - Implement a method called correct_name that will raise a NotImplementedError.
- Turn your Person class to Nameable
  - Make sure that your Person class inherits from Nameable
  - Make sure that this class has a method correct_name implemented. It should simply return the name attribute.
- Prepare base Decorator
  - Make sure that it inherits from Nameable.
  - In the constructor assign a nameable object from params to an instance variable.
  - Implement the correct_name method that returns the result of the correct_name method of the @nameable.
- Prepare CapitalizeDecorator and TrimmerDecorator
  - For the CapitalizeDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method correct_name that capitalizes the output of @nameable.correct_name.
  - For the TrimmerDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

- Result:
  image

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -13,9 +16,49 @@ def can_use_services?
...
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.upcase
```
Comment (FrederickMih) [inline person.rb:45]: - [Optional] This method(upcase) would transform a string to uppercase. Based on the requirements of this project, the CapitalizeDecorator method should capitalize the first letter of the given name.
- **Hint**: Let's try out `@nameable.correct_name.capitalize`

**Review by FrederickMih — APPROVED** [review]
Hi @AntonioHincapie , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (AntonioHincapie) [reply person.rb:45]: @FrederickMih Thanks for your feedback and recommendations.
Please can you explain this a little bit further, I was reading again the project requirements and it says that the output should be capitalized it doesn't say that it has to be only the first letter.
Thanks again for your feedback and hope to hear soon from you.

Comment (FrederickMih) [reply person.rb:45]: @AntonioHincapie, you don't need to transform the name to uppercase. Here you just need to capitalize the first letter of the word. When I'm asked to enter my name, I may mistakenly write `frederick` which is wrong. A name of a person is a proper noun which means that it needs to begin with a capital letter. The app should be able to correct **frederick** to **Frederick** hence capitalizing here means to take the first letter of the name uppercase.

Hope it's a bit clearer now.

**Pull request merged**